cognite/client/data_classes/data_modeling/query.py

```diff
@@ -159,6 +159,23 @@ def load_yaml(cls, data: str) -> Query:
         )
         return cls.load(data)
 
+    def _validate_for_query(self) -> None:
```

Comment:
Nice! I'm now regretting that we didn't use separate data classes for sync and query, but it is what it is. Let's add similar validation for passing `sort` when using /sync?